core/src/main/java/org/apache/calcite/sql/fun/SqlItemOperator.java

```diff
@@ -115,7 +115,8 @@ private SqlSingleOperandTypeChecker getChecker(SqlCallBinding callBinding) {
 
   @Override public String getAllowedSignatures(String name) {
     return "<ARRAY>[<INTEGER>]\n"
-        + "<MAP>[<VALUE>]";
+        + "<MAP>[<VALUE>]\n"
+        + "<ROW>[<CHARACTER>|<INTEGER>]";
   }
```

Comment:
Why we use the `VALUE` keyword before ?

Reply:
`VALUE` makes sense for `MAP` as it stands for the value for the datatype used for the key of the map itself (which is not known a priori), for `ROW`, very much like for `ARRAY`, we know the types for our index operand, so it's better to refer to them explicitly IMO.

Reply:
I mean `<MAP>[<CHARACTER>]` seems more straight-forward.

Reply:
Ah, sorry, now I get it. 

Consider that `MAP` is a shortcut for ` MAP<K,V>`, so `<MAP>[<VALUE>]` should be read as `<MAP>[<K>]`, since we will finally be accessing the `MAP` values via keys, which can have type other than `CHARACTER`(check [SqlItemOperator.java#101](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/fun/SqlItemOperator.java#L101), for instance).

This said, probably `<MAP>[<KEY>]` or even `<MAP>[<KEY_VALUE>]` seem more clear than `<MAP>[<VALUE>]`, I could change that if you agree on that.

Reply:
I think we should use the expected data type instead of the attribute name, so `<MAP>[< CHARACTER >]` seems better.

Reply:
`<MAP>[< CHARACTER >]` is only correct when we have a `MAP<CHARACTER, _>` collection, which is not always the case. 

That signature is static, so the concrete type is not known, the best we can do is to use a "placeholder" for the right type, being it `<VALUE>` or `<KEY>` or something else.

To make it clearer, consider that we could legally write a query like `select map_field[select ...] ...`, provided that the inner `select ...` returns an expression of type `K`, for a `map_field` of type `MAP<K, _>`.

Reply:
Thanks, `<MAP>[<KEY>]` seems more reasonable.

Reply:
In signatures, I believe the terms in angle brackets are always data types. So, `<MAP>[<ANY>]` would be appropriate. Take a look at other implementations of `SqlOperator.getAllowedSignatures(String)` to confirm.

Reply:
Yes, the other implementations of `SqlOperator.getAllowedSignatures(String)` consistently use data types, so I will adopt `<MAP>[<ANY>]` as suggested.